services:
  traefik:
    build:
      context: ./traefik
      args:
        - TRAEFIK_USER=${TRAEFIK_USER}
        - TRAEFIK_PASSWORD=${TRAEFIK_PASSWORD}
        - PASSWORD_FILE_PATH=/auth/usersfile
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./traefik/conf:/etc/traefik/
      - traefik_certs:/acme
    ports:
      - 80:80
      - 443:443
    restart: unless-stopped
    networks:
      - default
      - odoo-network
    extra_hosts:
      - "host.docker.internal:host-gateway"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.traefik.service=api@internal"
      - "traefik.http.routers.traefik.middlewares=auth@file"
      - "traefik.http.routers.traefik.tls.certresolver=letsencrypt"

  glances:
    image: nicolargo/glances
    volumes:
      - ./glances/glances.conf:/glances/conf/glances.conf
      - /var/run/docker.sock:/var/run/docker.sock:ro
    environment:
      - GLANCES_OPT=-C /glances/conf/glances.conf -w
    labels:
      - "traefik.enable=true"
      - "traefik.http.services.glances.loadbalancer.server.port=61208"
      - "traefik.http.routers.glances.middlewares=auth@file"
      - "traefik.http.routers.glances.tls.certresolver=letsencrypt"
    pid: host

  portainer:
    build:
      context: ./portainer
      args:
        - PORTAINER_ADMIN_PASSWORD=${PORTAINER_ADMIN_PASSWORD}
    restart: unless-stopped
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - portainer_data:/data
    labels:
      - "traefik.enable=true"
      - "traefik.http.services.portainer.loadbalancer.server.port=9000"
      - "traefik.http.routers.portainer.tls.certresolver=letsencrypt"

  ha:
    image: "ghcr.io/home-assistant/home-assistant:2025.7"
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - ./home-assistant/config:/config
    network_mode: host # Only for Linux, incompatible with Traefik but allows MDNS discovery
    restart: unless-stopped
    privileged: true
    environment:
      - LOCAL_IP=${LOCAL_IP}
    labels:
      - "traefik.enable=true"
      - "traefik.http.services.ha.loadbalancer.server.url=http://host.docker.internal:8123"
      - "traefik.http.routers.ha.tls.certresolver=letsencrypt"
    devices:
      - /dev/serial/by-id/${ZIGBEE_USB}:/dev/ttyUSB0
    profiles:
      - ha

  odoo:
    build:
      context: ./odoo
      args:
        - ODOO_MASTER_PASSWORD=${ODOO_MASTER_PASSWORD}
        - ODOO_DATABASE_NAME=${ODOO_DATABASE_NAME}
    depends_on:
      - odoo-db
    environment:
      - HOST=odoo-db
      - USER=${ODOO_DB_USER}
      - PASSWORD=${ODOO_DB_PASSWORD}
    profiles:
      - odoo
    volumes:
      - odoo_addons:/mnt/extra-addons
      - odoo_web_data:/var/lib/odoo
    networks:
      - odoo-network
    labels:
      - "traefik.enable=true"
      - "traefik.http.services.odoo.loadbalancer.server.port=8069"
      - "traefik.http.routers.odoo.tls.certresolver=letsencrypt"

  odoo-db:
    image: postgres
    profiles:
      - odoo
    environment:
      - POSTGRES_DB=${ODOO_DATABASE_NAME}
      - POSTGRES_USER=${ODOO_DB_USER}
      - POSTGRES_PASSWORD=${ODOO_DB_PASSWORD}
    volumes:
      - odoo_datas:/var/lib/postgresql/data
    networks:
      - odoo-network

volumes:
  traefik_certs:
  portainer_data:
  odoo_datas:
  odoo_web_data:
  odoo_addons:

networks:
  odoo-network:
  default:
    driver: bridge
    ipam:
      config:
        - subnet: ${LOCAL_IP}
